packages/core/database/lib/index.js

```diff
@@ -74,7 +74,12 @@ class Database {
 
 // TODO: move into strapi
 Database.transformContentTypes = transformContentTypes;
-Database.init = async (config) => new Database(config);
+Database.init = async (config) => {
+  const instance = new Database(config);
+  instance.databaseInformation = await instance.dialect.schemaInspector.getDatabaseInformation();
+
+  return instance;
+};
```

Comment:
I'm not a fan of this. Is there a better way?

Reply:
It doesn't look bad, enhancing the instance object with a valuable property. Why don't you like it?

Reply:
I know what you can do.
You can add databaseInformation in Database. Or add an async method in Database Class that will return `return this.dialect.schemaInspector.getDatabaseInformation()`.

Reply:
If you're fine with that I'm fine with it too :p I wanted to have it in the constructor but I cannot since it's async.

It's what I did at the beginning but it does a query to the database each time then. I would like to avoid it as it can happen to use databaseInformation several times when editing relations.

Or I can do some kind of "singleton".
```js
Database.getDatabaseInformation();

getDatabaseInformation() {
  if (!this.databaseInformation) {
    this.databaseInformation = this.dialect.schemaInspector.getDatabaseInformation();
  }
  return this.databaseInformation;
}
```

Reply:
(i did it and pushed it)